src/runtime/vm/executable.cc

```diff
@@ -449,16 +448,11 @@ void Executable::LoadConstantSection(dmlc::Stream* stream) {
       STREAM_CHECK(stream->Read(&name), "late-bound constant name");
       constants[const_index] = NDArray(nullptr);
       late_bound_constant_names[const_index] = std::move(name);
-      any_late_bound = true;
     } else {
       STREAM_CHECK(false, "constant tag");
     }
   }
 
-  if (!any_late_bound) {
-    late_bound_constant_names.clear();
```

Comment:
LGTM. Not sure why I was worried about disallowing LoadLateBoundConstants if there are no late-bound constants. Sorry about that.